### PR TITLE
planner: fix bug when pruning columns for TableDual

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve? 

Dual may not be from only one data source.
So check `cols[0]` is not enough.

### What is changed and how it works?

Check the full slice.

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: planner/core/rule_column_pruning.go [hunk 1]
```diff
@@ -262,8 +262,15 @@ func (p *LogicalTableDual) PruneColumns(parentUsedCols []*expression.Column) err
 		}
 	}
 	for k, cols := range p.schema.TblID2Handle {
-		if p.schema.ColumnIndex(cols[0]) == -1 {
+		for i := len(cols) - 1; i >= 0; i-- {
```
Comment (eurekaka) [inline planner/core/rule_column_pruning.go]: The content of `TblID2Handle` looks weird for the above test case, i.e, `t1.a` and `t2.a` are all stored in single bucket of TblID2Handle[70] in my instance for example, though `t1` and `t2` are tow aliases for same table `t`, actually they should be 2 separate tables in the query. Would it have risk in executor when using `TblID2Handle` like this?

Comment (lzmhhh123) [reply planner/core/rule_column_pruning.go]: Is any situation that the handle has not only one column? I think `TblID2Handle` should be the `map<id, *Column>`.

Comment (lzmhhh123) [reply planner/core/rule_column_pruning.go]: Anyway. This is one of the chaos of `Schema`. Maybe temporarily fix bugs is OK.

Comment (winoros) [reply planner/core/rule_column_pruning.go]: @lzmhhh123 Multi delete will map to multiple Column.

**Comment by winoros** [comment]
@lzmhhh123 Currently, no. We can move this thing out of schema ASAP.

**Pull request merged**